### PR TITLE
APIS-6893 - Integration and component tests moved to sub-projects

### PR DESCRIPTION
**Pull request by peteslater-ee**

### CODE REVIEW

**Comment by peteslater-ee** [comment]
test this please

**Review by Tobias1087 — APPROVED** [review]

File: build.sbt [hunk 1]
```diff
@@ -113,14 +94,32 @@ lazy val microservice = Project(appName, file("."))
...
-  Command.command("run-all-tests") { state => "test" :: "it:test" :: "component:test" :: state },
 
-  Command.command("clean-and-test") { state => "clean" :: "compile" :: "run-all-tests" :: state },
+lazy val it = (project in file("it"))
+  .enablePlugins(PlayScala)
+  .dependsOn(microservice % "test->test")
+  .settings(
+    name := "integration-tests",
+    Test / testOptions += Tests.Argument(TestFrameworks.ScalaTest, "-eT"),
+    DefaultBuildSettings.itSettings()
```
Comment (johnsgp) [inline build.sbt:104]: Do we need this here?
```
    addTestReportOption(Test, "int-test-reports")
```

File: build.sbt [hunk 2]
```diff
@@ -113,14 +94,32 @@ lazy val microservice = Project(appName, file("."))
...
-  Command.command("pre-commit") { state => "clean" :: "scalafmtAll" :: "scalafixAll" :: "coverage" :: "testOnly * -- -l ExcludeFromCoverage" :: "it:test" :: "component:test" :: "coverageOff" :: "testOnly * -- -n ExcludeFromCoverage" :: "coverageReport" :: state }
+lazy val component = (project in file("component"))
+  .enablePlugins(PlayScala)
+  .dependsOn(microservice % "test->test")
+  .settings(
+    name := "component-tests",
+    libraryDependencies ++= AppDependencies.compile ++ AppDependencies.test ++ AppDependencies.componentTestDependencies,
+    Test / testOptions := Seq(Tests.Argument(TestFrameworks.ScalaTest, "-eT")),
+    DefaultBuildSettings.itSettings()
+  )
```
Comment (johnsgp) [inline build.sbt:115]: Do we need this here?
```
    addTestReportOption(Test, "component-reports")
```

File: build.sbt [hunk 3]
```diff
@@ -18,11 +18,11 @@ lazy val appName = "third-party-developer-frontend"
...
-
-ThisBuild / libraryDependencySchemes += "org.scala-lang.modules" %% "scala-xml" % VersionScheme.Always
+ThisBuild / scalaVersion := "2.13.12"
+ThisBuild / majorVersion := 0
 ThisBuild / semanticdbEnabled := true
 ThisBuild / semanticdbVersion := scalafixSemanticdb.revision
+ThisBuild / libraryDependencySchemes += "org.scala-lang.modules" %% "scala-xml" % VersionScheme.Always
 
 lazy val microservice = Project(appName, file("."))
   .enablePlugins(plugins: _*)
```
Comment (johnsgp) [inline build.sbt:28]: Could we add this here?
```
  .disablePlugins(JUnitXmlReportPlugin)
```

File: build.sbt [hunk 4]
```diff
@@ -113,14 +94,32 @@ lazy val microservice = Project(appName, file("."))
...
 
-  // Coverage does not need compile !
-  Command.command("pre-commit") { state => "clean" :: "scalafmtAll" :: "scalafixAll" :: "coverage" :: "testOnly * -- -l ExcludeFromCoverage" :: "it:test" :: "component:test" :: "coverageOff" :: "testOnly * -- -n ExcludeFromCoverage" :: "coverageReport" :: state }
+lazy val component = (project in file("component"))
+  .enablePlugins(PlayScala)
+  .dependsOn(microservice % "test->test")
+  .settings(
+    name := "component-tests",
+    libraryDependencies ++= AppDependencies.compile ++ AppDependencies.test ++ AppDependencies.componentTestDependencies,
+    Test / testOptions := Seq(Tests.Argument(TestFrameworks.ScalaTest, "-eT")),
```
Comment (johnsgp) [inline build.sbt:113]: In other repos we have `+=` as in
```suggestion
    Test / testOptions += Seq(Tests.Argument(TestFrameworks.ScalaTest, "-eT")),
```

File: build.sbt [hunk 5]
```diff
@@ -60,25 +60,6 @@ lazy val microservice = Project(appName, file("."))
     Test / unmanagedSourceDirectories ++= Seq(baseDirectory.value / "test", baseDirectory.value / "test-utils"),
```
Comment (johnsgp) [inline build.sbt:60]: Do we need to state "test" explicitly as a source directory? Elsewhere it's added by default. I realise this is from past work.

Comment (peteslater-ee) [reply build.sbt:60]: Nope

File: build.sbt (hunk 4 above)
Comment (peteslater-ee) [reply build.sbt:113]: That's how I found it so haven't changed it.

Comment (peteslater-ee) [reply build.sbt:113]: Also you will note that in some places we set the value to a Seq whereas in others we add a value to the existing Seq.
I may changed them all to add to the Seq for consistency

File: build.sbt (hunk 1 above)
Comment (peteslater-ee) [reply build.sbt:104]: I really don't know

**Review by johnsgp — APPROVED** [review]

**Pull request merged**